src/danog/MadelineProto/AnnotationsBuilder.php

```diff
@@ -0,0 +1,137 @@
+<?php
+/*
+ Copyright 2016 Daniil Gentili
+(https://daniil.it)
+This file is part of MadelineProto.
+MadelineProto is free software: you can redistribute it and/or modify it under the terms of the GNU Affero General Public License as published by the Free Software Foundation, either version 3 of the License, or (at your option) any later version.
+MadelineProto is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+See the GNU Affero General Public License for more details.
+You should have received a copy of the GNU General Public License along with MadelineProto.
+If not, see <http://www.gnu.org/licenses/>.
+*/
+
+namespace danog\MadelineProto;
+
+use phpDocumentor\Reflection\DocBlockFactory;
+
+class AnnotationsBuilder
+{
+    use \danog\MadelineProto\TL\TL;
+
+    public function __construct($settings)
+    {
+        $this->construct_TL($settings['tl_schema']);
+        $this->settings = $settings;
+    }
+
+    public function mk_annotations()
+    {
+        \danog\MadelineProto\Logger::log('Generating annotations...');
+        $this->setProperties();
+        $this->createInternalClasses();
+    }
+
+    /**
+     * Open file of class APIFactory
+     * Insert properties
+     * save the file with new content.
+     */
+    private function setProperties()
+    {
+        \danog\MadelineProto\Logger::log('Generating properties...');
+        $fixture = DocBlockFactory::createInstance();
+        $class = new \ReflectionClass(APIFactory::class);
+        $content = file_get_contents($filename = $class->getFileName());
+        foreach ($class->getProperties() as $property) {
+            if ($raw_docblock = $property->getDocComment()) {
+                $docblock = $fixture->create($raw_docblock);
+                if ($docblock->hasTag('internal')) {
+                    $content = str_replace("\n    ".$raw_docblock."\n    public $".$property->getName().';', '', $content);
+                }
+            }
+        }
+        foreach ($this->methods->method_namespace as $namespace) {
+            $content = preg_replace(
+                '/(class( \w+[,]?){0,}\n{\n)/',
+                '${1}'.
+                "    /**\n".
+                "     * @internal this is a internal property generated by build_docs.php, dont change manually\n".
```

Comment:
Dont is missing an apostrophe, it should be don't since it's an abbreviation of do not (yes I'm nitpicking, this pr is so nice I wanted to make it perfect).